### PR TITLE
fix mach spawn fail interrupt

### LINKED ISSUE #17626: OSX Travis fails spawn test with InterruptException
e.g. https://travis-ci.org/JuliaLang/julia/jobs/147328890
or https://travis-ci.org/JuliaLang/julia/jobs/146808733

```
    From worker 6:       * spawn                       [stdio passthrough ok]
    From worker 6:  Error During Test
    From worker 6:    Test threw an exception of type InterruptException
    From worker 6:    Expression: success(truecmd)
    From worker 6:    InterruptException:
    From worker 6:     in push! at ./array.jl:444 [inlined]
    From worker 6:     in wait(::Condition) at ./event.jl:14
    From worker 6:     in stream_wait(::Base.Process, ::Condition, ::Vararg{Condition,N}) at ./stream.jl:44
    From worker 6:     in wait at ./process.jl:712 [inlined]
    From worker 6:     in success(::Base.Process) at ./process.jl:602
    From worker 6:     in success(::Cmd) at ./process.jl:608
    From worker 6:     in include_string(::String, ::String) at ./loading.jl:380
    From worker 6:     in include_from_node1(::String) at ./loading.jl:429
    From worker 6:     in macro expansion at ./util.jl:180 [inlined]
    From worker 6:     in runtests(::String) at /private/tmp/julia/share/julia/test/testdefs.jl:7
    From worker 6:     in (::Base.Serializer.__deserialized_types__.##277)(::String) at /private/tmp/julia/share/julia/test/runtests.jl:44
    From worker 6:     in (::Base.##499#501{Base.CallMsg{:call_fetch}})() at ./multi.jl:1193
    From worker 6:     in run_work_thunk(::Base.##499#501{Base.CallMsg{:call_fetch}}, ::Bool) at ./multi.jl:844
    From worker 6:     in macro expansion at ./multi.jl:1193 [inlined]
    From worker 6:     in (::Base.##498#500{Base.CallMsg{:call_fetch},Base.MsgHeader,TCPSocket})() at ./event.jl:46
```

### PR DESCRIPTION
**Pull request by vtjnash**
travis seemed to be having issues with the cache in the previous PR (https://github.com/JuliaLang/julia/pull/19590), so attempting a new one.

edit by tkelman: ~~should close https://github.com/JuliaLang/julia/issues/17626~~ sadly that was over-optimistic

### CODE REVIEW

File: test/spawn.jl [hunk 1]
```diff
@@ -64,12 +64,12 @@ if valgrind_off
 end
 
 if is_unix()
-    prefixer(prefix, sleep) = `perl -nle '$|=1; print "'$prefix' ", $_; sleep '$sleep';'`
-    @test success(pipeline(`perl -le '$|=1; for(0..2){ print; sleep 1 }'`,
-                       prefixer("A",2) & prefixer("B",2)))
-    @test success(pipeline(`perl -le '$|=1; for(0..2){ print; sleep 1 }'`,
-                       prefixer("X",3) & prefixer("Y",3) & prefixer("Z",3),
-                       prefixer("A",2) & prefixer("B",2)))
+    prefixer(prefix, sleep) = `sh -c "while read; do echo '$prefix ' \$REPLY; sleep $sleep; done"`
```
Comment (tkelman) [inline test/spawn.jl]: is REPLY going to work in all posix shells?

Comment (vtjnash) [inline test/spawn.jl]: hm, looks like specifying it explicitly might be more compatible (with dash and ash)

File: .travis.yml [hunk 2]
```diff
@@ -102,7 +97,7 @@ script:
     # capture the log, but only print it if `make deps` fails
     # try to show the end of the log first, because this log might be very long (> 4MB)
     # and thus be truncated by travis
-    - moreutils/mispipe "make $BUILDOPTS NO_GIT=1 -C deps 2> deps-err.log" "$BAR" > deps.log ||
+    - moreutils/mispipe "make \$BUILDOPTS NO_GIT=1 -C deps 2> deps-err.log" "$BAR" > deps.log ||
```
Comment (tkelman) [inline .travis.yml]: Ok I'm at least convinced that this is fine on linux - not working yet on mac though. Comment out the linux entries of the matrix while debugging that?

Comment (vtjnash) [inline .travis.yml]: Trying to figure out why mac wouldn't build this PR is why I started down the path of trying to see if the failure was caused by this mis-setting of BUILDOPTS.

Comment (tkelman) [inline .travis.yml]: And I've been saying, the only difference this backslash makes is in the spawn bit that doesn't matter for the deps

Comment (vtjnash) [inline .travis.yml]: It slightly changes how the command get rewritten and parsed. But anyways, this commit didn't use to pass on it's own with the same failure.

Comment (tkelman) [inline .travis.yml]: has this ever passed on mac travis? the libuv changes have always caused it to freeze as far as I can tell

**Comment by iamed2** [comment]
Would this test failure be an example of this?: https://gist.github.com/iamed2/e0a7fda816a5f8bb3d65f0124c70bb78

**Comment by vtjnash** [comment]
yes.

**Review by tkelman — APPROVED** [review]
umpteenth time's the charm. lgtm unless the currently running win64 appveyor job shows anything surprising

**Comment by vtjnash** [comment]
yeah, it's no fun running into mach bugs trying to avoid other ones :/

**Pull request merged**